<div class="row mb-3">
  <div class="col-12">
    <h3 class="d-inline-block align-middle">
      Variant Functional Prediction Scores of
      Chr{{ variant.chr }}:{{ variant.pos }} {{ variant.ref }}&gt;{{ variant.alt }}
      from dbNSFP
    </h3>
  </div>

  <div class="col-12 mt-3" *ngIf="!loading && data && rankAverage != null" [@toggle]>
    <div class="d-table comment">
      <div class="d-table-cell align-top comment-icon-wrapper">
        <mat-icon matTooltip="MARRVEL's comment" matTooltipPosition="above">insert_comment</mat-icon>
      </div>

      <span class="d-table-cell pl-2">
        For Chr{{ variant.chr }}:{{ variant.pos }} {{ variant.ref }}&gt;{{ variant.alt }},
        the average dbNSFP rank score is {{ rankAverage.toFixed(3) }}.
        This means that Chr{{ variant.chr }}:{{ variant.pos }} {{ variant.ref }}&gt;{{ variant.alt }}
        is <strong>more likely to be damaging
        than {{ (rankAverage * 100).toFixed(1) }}% of all possible coding variants</strong>.
      </span>
    </div>
  </div>
</div>

<div class="row" *ngIf="!loading && !data">
  <div class="col-12">
    <p class="m-0">No data found for Chr{{ variant.chr }}:{{ variant.pos }} {{ variant.ref }}&gt;{{ variant.alt }}</p>
  </div>
</div>

<div class="row" *ngIf="!loading && data">
  <div class="col-12">
    <table class="mat-table w-100">
      <thead>
        <tr class="mat-header-row">
          <th class="mat-header-cell">
            Prediction tool
          </th>
          <th class="mat-header-cell number nowrap-fit-cell">
            Score / Prediction
          </th>
          <th class="mat-header-cell">
            <span class="text-underline-dashed" matTooltip="The Rank Score is provided by dbNSFP to allow comparison between different scores. It is between 0 and 1 and a score of 0.9 means it is more likely to be damaging than 90% of all potential nsSNVs predicted by that method.">
              dbNSFP Rank Score
            </span>
          </th>
        </tr>
      </thead>

      <tbody>
        <tr class="mat-row">
          <td class="mat-cell">
            <span class="text-underline-dashed">
              CADD phred
            </span>
            <span class="ml-1 badge badge-secondary">v1.7</span>
          </td>
          <td class="mat-cell number">
            <span class="text-underline-dashed"
              *ngIf="data.scores.CADD && data.scores.CADD.phred"
              matTooltip="The larger the score the more likely the SNP has damaging effect. "
            >
              {{ data.scores.CADD.phred }}&nbsp;
            </span>
            <span class="text-muted" *ngIf="!data.scores.CADD || !data.scores.CADD.phred">NA</span>
          </td>
          <td class="mat-cell">
            <app-rankscore-visual
              *ngIf="data.scores.CADD && data.scores.CADD.phred"
              [rankscore]="data.scores.CADD.rankscore"
            ></app-rankscore-visual>
          </td>
        </tr>

        <tr class="mat-row">
          <td class="mat-cell">
            <span class="text-underline-dashed"
              matTooltip="Ensemble method of 13 tools.">
              REVEL
              <span class="ml-1 badge badge-secondary">
                May 3, 2021
              </span>
            </span>
          </td>
          <td class="mat-cell number">
            <span class="text-underline-dashed na" *ngIf="data.scores.REVEL && data.scores.REVEL.scores"
              matTooltip="Range = 0 (least damaging) to 1 (most damaging)"
            >{{ max(data.scores.REVEL.scores) }}</span>
          </td>
          <td class="mat-cell">
            <app-rankscore-visual
              *ngIf="data.scores.REVEL && data.scores.REVEL.scores"
              [rankscore]="data.scores.REVEL.rankscore"
            ></app-rankscore-visual>
          </td>
        </tr>

        <tr class="mat-row">
          <td class="mat-cell">
            <span class="text-underline-dashed"
              matTooltip="unsupervised model incorporating structural context of an AlphaFold-derived system predicts the pathogenicity of human missense variants.">
              AlphaMissense
            </span>
          </td>
          <td class="mat-cell number">
            <span class="text-underline-dashed number na"
              *ngIf="data.scores.AlphaMissense && data.scores.AlphaMissense.predictions"
              matTooltip="Range: 0 to 1. The larger the score, the more likely the variant is pathogenic.">{{ maxStr(data.scores.AlphaMissense.predictions, 'AlphaMissense') }}</span>
          </td>
          <td class="mat-cell">
            <app-rankscore-visual
              *ngIf="data.scores.AlphaMissense && data.scores.AlphaMissense.predictions"
              [rankscore]="data.scores.AlphaMissense.rankscore"
            ></app-rankscore-visual>
          </td>
        </tr>

        <tr class="mat-row">
          <td class="mat-cell">
            <span class="text-underline-dashed"
              matTooltip="Pathogenicity prediction for missense variants based on common variants of non-human primate species using a deep neural network."
            >PrimateAI</span>
          </td>
          <td class="mat-cell number">
            <span class="text-underline-dashed"
              *ngIf="data.scores.PrimateAI && data.scores.PrimateAI.prediction"
              matTooltip="Possible scores are: Tolerated or Damaging">
              {{ getPredLabel(data.scores.PrimateAI.prediction, 'PrimateAI') }}
            </span>
            <span class="text-muted" *ngIf="!data.scores.PrimateAI || !data.scores.PrimateAI.prediction">NA</span>
          </td>
          <td class="mat-cell">
            <app-rankscore-visual
              *ngIf="data.scores.PrimateAI && data.scores.PrimateAI.prediction"
              [rankscore]="data.scores.PrimateAI.rankscore"
            ></app-rankscore-visual>
          </td>
        </tr>

        <tr class="mat-row">
          <td class="mat-cell">
            <span class="text-underline-dashed"
              matTooltip="MutationTaster prediction. The score cutoff between Disease Causing and Polymorphism is 0.5 for MTnew and 0.31733 for the rankscore."
            >
              MutationTaster
              <span class="ml-1 badge badge-secondary">2021</span>
            </span>
          </td>
          <td class="mat-cell number">
            <span class="text-underline-dashed na"
              *ngIf="data.scores.MutationTaster && data.scores.MutationTaster.predictions"
              matTooltip="Possible scores are: Disease Causing Automatic, Disease Causing, Polymorphism, and Polymorphism Automatic">{{ maxStr(data.scores.MutationTaster.predictions, 'MutationTaster') }}</span>
          </td>
          <td class="mat-cell">
            <app-rankscore-visual
              *ngIf="data.scores.MutationTaster && data.scores.MutationTaster.predictions"
              [rankscore]="data.scores.MutationTaster.convertedRankscore"
            ></app-rankscore-visual>
          </td>
        </tr>

        <tr class="mat-row">
          <td class="mat-cell">
            <span class="text-underline-dashed"
              matTooltip="Uses conservation data and trained on mutations linked to Mendelian diseases."
            >
              M-CAP
              <span class="ml-1 badge badge-secondary">v1.3</span>
            </span>
          </td>
          <td class="mat-cell number na">
            <span class="text-underline-dashed" *ngIf="data.scores.MCAP && data.scores.MCAP.prediction"
              matTooltip="Possible scores are: Tolerated or Damaging">{{ getPredLabel(data.scores.MCAP.prediction, 'MCAP') }}</span>
          </td>
          <td class="mat-cell">
            <app-rankscore-visual
              *ngIf="data.scores.MCAP && data.scores.MCAP.prediction"
              [rankscore]="data.scores.MCAP.rankscore"
            ></app-rankscore-visual>
          </td>
        </tr>

        <tr class="mat-row">
          <td class="mat-cell">
            <span class="text-underline-dashed"
              matTooltip="Uses eight sequence-based and three structure-based predictive features. Trained with Mendelian disease mutations and SNVs from close mammalian homolog proteins."
            >
              Polyphen-2 HumDiv
              <span class="ml-1 badge badge-secondary">v2.2.2</span>
            </span>
          </td>
          <td class="mat-cell number">
            <span class="text-underline-dashed na"
              *ngIf="data.scores.Polyphen2HDIV && data.scores.Polyphen2HDIV.predictions"
              matTooltip="Possible scores are: Benign, Possibly Damaging, and Probably Damaging">{{ maxStr(data.scores.Polyphen2HDIV.predictions, 'Polyphen2HDIV') }}</span>
          </td>
          <td class="mat-cell">
            <app-rankscore-visual
              *ngIf="data.scores.Polyphen2HDIV && data.scores.Polyphen2HDIV.predictions"
              [rankscore]="data.scores.Polyphen2HDIV.rankscore"
            ></app-rankscore-visual>
          </td>
        </tr>

        <tr class="mat-row">
          <td class="mat-cell">
            <span class="text-underline-dashed"
              matTooltip="Uses eight sequence-based and three structure-based predictive features. Trained with disease associated and common SNVs."
            >
              Polyphen-2 HumVar
              <span class="ml-1 badge badge-secondary">v2.2.2</span>
            </span>
          </td>
          <td class="mat-cell number">
            <span class="text-underline-dashed number na"
              *ngIf="data.scores.Polyphen2HVAR && data.scores.Polyphen2HVAR.predictions"
              matTooltip="Possible scores are: Benign, Possibly Damaging, and Probably Damaging">{{ maxStr(data.scores.Polyphen2HVAR.predictions, 'Polyphen2HVAR') }}</span>
            <span class="text-muted" *ngIf="!data.scores.Polyphen2HVAR || !data.scores.Polyphen2HVAR.predictions">NA</span>
          </td>
          <td class="mat-cell">
            <app-rankscore-visual
              *ngIf="data.scores.Polyphen2HVAR && data.scores.Polyphen2HVAR.predictions"
              [rankscore]="data.scores.Polyphen2HVAR.rankscore"
            ></app-rankscore-visual>
          </td>
        </tr>

        <tr class="mat-row">
          <td class="mat-cell">
            <span class="text-underline-dashed"
              matTooltip="Uses multiple alignments and phylogenetic tree of 34 mammals."
            >
              GERP++
            </span>
          </td>
          <td class="mat-cell number">
            <span class="text-underline-dashed number"
              *ngIf="data.scores['GERP++RS'] && data.scores['GERP++RS'].score"
              matTooltip="-12.3 (least conserved) to 6.17 (most conserved)">
              {{ toFixed(data.scores['GERP++RS'].score, 3) }}
            </span>
            <span class="text-muted number" *ngIf="!data.scores['GERP++RS'] || !data.scores['GERP++RS'].score">NA</span>
          </td>
          <td class="mat-cell">
            <app-rankscore-visual
              *ngIf="data.scores['GERP++RS'] && data.scores['GERP++RS'].score"
              [rankscore]="data.scores['GERP++RS'].rankscore"
            ></app-rankscore-visual>
          </td>
        </tr>

        <tr class="mat-row">
          <td class="mat-cell">
            <span class="text-underline-dashed"
              matTooltip="Uses multiple alignments and phylogenetic tree of 100 vertebrates."
            >
              phyloP 100way Vertebrate
            </span>
          </td>
          <td class="mat-cell number">
            <span class="text-underline-dashed"
              *ngIf="data.scores.phyloP100wayVertebrate && data.scores.phyloP100wayVertebrate.score"
              matTooltip="Range =- 20.0 (least conserved) to 10.003 (most conserved)"
            >
              {{ data.scores.phyloP100wayVertebrate.score }}
            </span>
            <span class="text-muted" *ngIf="!data.scores.phyloP100wayVertebrate || !data.scores.phyloP100wayVertebrate.score">NA</span>
          </td>
          <td class="mat-cell">
            <app-rankscore-visual
              *ngIf="data.scores.phyloP100wayVertebrate && data.scores.phyloP100wayVertebrate.score"
              [rankscore]="data.scores.phyloP100wayVertebrate.rankscore"
            ></app-rankscore-visual>
          </td>
        </tr>

        <tr class="mat-row">
          <td class="mat-cell">
            <span class="text-underline-dashed"
              matTooltip="conservation score based on the multiple alignments of 470 mammalian genomes (including human)."
            >
                phyloP 470way Mammalian
            </span>
          </td>
          <td class="mat-cell number">
            <span class="text-underline-dashed"
              *ngIf="data.scores.phyloP470wayMammalian && data.scores.phyloP470wayMammalian.score"
              matTooltip="Scores range from -20 to 11.936. The larger the score, the more conserved the site."
            >
              {{ data.scores.phyloP470wayMammalian.score }}
            </span>
            <span class="text-muted" *ngIf="!data.scores.phyloP470wayMammalian || !data.scores.phyloP470wayMammalian.score">NA</span>
          </td>
          <td class="mat-cell">
            <app-rankscore-visual
              *ngIf="data.scores.phyloP470wayMammalian && data.scores.phyloP470wayMammalian.score"
              [rankscore]="data.scores.phyloP470wayMammalian.rankscore"
            ></app-rankscore-visual>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
