<app-navbar></app-navbar>

<div class="container mt-5 pt-5" *ngIf="!hideUpBox" [@toggle]>
  <div class="row">
    <div class="col-sm-8 offset-sm-2 col-md-6 offset-md-3 col-xs-12 text-center">
      <img src="../../../../assets/img/marrvel.svg" class="w-75">
    </div>

    <div class="col-12 mt-5 text-left">
      <mat-radio-group color="primary" aria-label="Select an option" [(ngModel)]="selectedInputType" (change)="onInputTypeChange($event)" name="inputType">
        <mat-radio-button color="primary" value="gene" class="mr-2">Human gene</mat-radio-button>
        <mat-radio-button color="primary" value="protein" class="mr-2">Human protein</mat-radio-button>
      <mat-radio-button color="primary" value="modelgene" class="mr-2">Model organism gene</mat-radio-button>
        <mat-radio-button color="primary" value="vcf" class="mr-2">Upload VCF file</mat-radio-button>
        <mat-radio-button color="primary" value="multigenes" class="mr-2" selected>Multiple human genes</mat-radio-button>
      </mat-radio-group>
    </div>

    <div class="col-12 mt-2">
      <app-multiple-gene-box
        (searchClick)="search($event)"
      ></app-multiple-gene-box>
    </div>
  </div>
</div>

<div class="container-fluid py-5 px-md-5" *ngIf="genes && genes.length > 0" [@toggle]>
  <div class="row">
    <div class="col-12">
      <a class="e2e-trusted-url"
        role="button"
        [href]="tsvPageDownloadUrl"
        mat-flat-button color="primary"
        [disabled]="loading || !tsvPageDownloadUrl"
      >
        <mat-icon class="icon-18px mr-1 spin" *ngIf="loading || !tsvPageDownloadUrl">autorenew</mat-icon>
        <mat-icon class="icon-18px mr-1" *ngIf="!loading && tsvPageDownloadUrl">save_alt</mat-icon>
        Export this page (.tsv)
      </a>
      <button class="ml-1"
        mat-flat-button color="primary"
        *ngIf="!tsvWholeDownloadUrl"
        (click)="createDownloadUrl('.tsv', true)"
      >
        <mat-icon class="icon-18px mr-1">cloud_download</mat-icon>
        Start download all (.tsv)
      </button>
      <a class="ml-1 e2e-trusted-url" role="button"
        mat-flat-button color="accent"
        [href]="tsvWholeDownloadUrl"
        *ngIf="tsvWholeDownloadUrl"
        [@toggle]
      >
        <mat-icon class="icon-18px mr-1">save_alt</mat-icon>
        Download all (.tsv)
      </a>
    </div>

    <div class="col-12 mt-3" *ngIf="wholeLoading" [@toggle]>
      <h5 class="mb-1 text-left">Fetching data from server...</h5>
      <mat-progress-bar mode="buffer"
        [value]="wholeGenesHaveData / genes.length * 100"
      ></mat-progress-bar>
    </div>

    <div class="col-12 mt-3">
      <mat-paginator
        [length]="genes.length"
        [pageSize]="gPerPage"
        [pageSizeOptions]="[10, 30, 50]"
        [pageIndex]="curPage"
        (page)="onPageChange($event)"
      ></mat-paginator>

      <mat-table class="w-100" [dataSource]="dataSource">
        <ng-container matColumnDef="symbol">
          <mat-header-cell *matHeaderCellDef>Gene</mat-header-cell>
          <mat-cell *matCellDef="let element" class="break-word">
            <a [attr.href]="'/human/gene/' + element.entrezId" target="_blank">{{ element.symbol }}</a>
          </mat-cell>
        </ng-container>

        <ng-container matColumnDef="omimPhenotypes">
          <mat-header-cell *matHeaderCellDef class="number border-left">
            <span class="text-underline-dashed" matTooltip="OMIM number of phenotypes">
              # Phenotypes
            </span>
          </mat-header-cell>
          <mat-cell *matCellDef="let element" class="number border-left">
            <span *ngIf="element.omim">
              {{ element.omim.numPhenos || 0 }}
            </span>
            <span *ngIf="!element.omim" class="na">NA</span>
          </mat-cell>
        </ng-container>

        <ng-container matColumnDef="omimAllele">
          <mat-header-cell *matHeaderCellDef class="number">
            <span class="text-underline-dashed" matTooltip="OMIM number of alleric variants">
              # Variants
            </span>
          </mat-header-cell>
          <mat-cell *matCellDef="let element" class="number">
              <span *ngIf="element.omim">
                {{ element.omim.numVars || 0 }}
              </span>
              <span *ngIf="!element.omim" class="na">NA</span>
            </mat-cell>
        </ng-container>

        <ng-container matColumnDef="clinvarP">
          <mat-header-cell *matHeaderCellDef class="number border-left">
            <span class="text-underline-dashed" matTooltip="ClinVar number of alleles reported as pathogenic">
              # Pathogenic
            </span>
          </mat-header-cell>
          <mat-cell *matCellDef="let element" class="number border-left">
            <span *ngIf="element.clinvar">
              {{ element.clinvar.pathogenic || 0 }}
            </span>
            <span *ngIf="!element.clinvar" class="na">NA</span>
          </mat-cell>
        </ng-container>

        <ng-container matColumnDef="clinvarLP">
          <mat-header-cell *matHeaderCellDef class="number">
            <span class="text-underline-dashed" matTooltip="ClinVar number of alleles reported as likely pathogenic">
              # Likely Pathogenic
            </span>
          </mat-header-cell>
          <mat-cell *matCellDef="let element" class="number">
            <span *ngIf="element.clinvar">
              {{ element.clinvar.likelyPathogenic || 0 }}
            </span>
            <span *ngIf="!element.clinvar" class="na">NA</span>
          </mat-cell>
        </ng-container>

        <ng-container matColumnDef="clinvarLB">
          <mat-header-cell *matHeaderCellDef class="number">
            <span class="text-underline-dashed" matTooltip="ClinVar number of alleles reported as likely benign">
              # Likely Benign
            </span>
          </mat-header-cell>
          <mat-cell *matCellDef="let element" class="number">
            <span *ngIf="element.clinvar">
              {{ element.clinvar.likelyBenign || 0 }}
            </span>
            <span *ngIf="!element.clinvar" class="na">NA</span>
          </mat-cell>
        </ng-container>

        <ng-container matColumnDef="clinvarB">
          <mat-header-cell *matHeaderCellDef class="number">
            <span class="text-underline-dashed" matTooltip="ClinVar number of alleles reported as benign">
              # Benign
            </span>
          </mat-header-cell>
          <mat-cell *matCellDef="let element" class="number">
            <span *ngIf="element.clinvar">
              {{ element.clinvar.benign || 0 }}
            </span>
            <span *ngIf="!element.clinvar" class="na">NA</span>
          </mat-cell>
        </ng-container>

        <ng-container matColumnDef="g2mpHom">
          <mat-header-cell *matHeaderCellDef class="number border-left">
            <span class="text-underline-dashed" matTooltip="Homozygous count"># Hom</span>
          </mat-header-cell>
          <mat-cell *matCellDef="let element" class="number border-left">
            <span *ngIf="element.geno2mp">
              {{ element.geno2mp.homCounts }}
            </span>
            <span *ngIf="!element.geno2mp" class="na">NA</span>
          </mat-cell>
        </ng-container>

        <ng-container matColumnDef="g2mpHet">
          <mat-header-cell *matHeaderCellDef class="number">
            <span class="text-underline-dashed" matTooltip="Homozygous count"># Het</span>
          </mat-header-cell>
          <mat-cell *matCellDef="let element" class="number">
            <span *ngIf="element.geno2mp">
              {{ element.geno2mp.hetCounts }}
            </span>
            <span *ngIf="!element.geno2mp" class="na">NA</span>
          </mat-cell>
        </ng-container>

        <ng-container matColumnDef="g2mpHpo">
          <mat-header-cell *matHeaderCellDef class="number">
            <span class="text-underline-dashed" matTooltip="Number of HPO profiles">
              # HPO
            </span>
          </mat-header-cell>
          <mat-cell *matCellDef="let element" class="number">
            <span *ngIf="element.geno2mp">
              {{ element.geno2mp.hpoCounts }}
            </span>
            <span *ngIf="!element.geno2mp" class="na">NA</span>
          </mat-cell>
        </ng-container>

        <ng-container matColumnDef="gnomadSynZ">
          <mat-header-cell *matHeaderCellDef class="number border-left">
            <span class="text-underline-dashed" matTooltip="Synonymouse Z">
              Syn Z
            </span>
          </mat-header-cell>
          <mat-cell *matCellDef="let element" class="number border-left">
            <span *ngIf="element.gnomad && element.gnomad.syn">
              {{ element.gnomad.syn.z.toFixed(2) }}
            </span>
            <span *ngIf="!element.gnomad || !element.gnomad.syn" class="na">NA</span>
          </mat-cell>
        </ng-container>

        <ng-container matColumnDef="gnomadMisZ">
          <mat-header-cell *matHeaderCellDef class="number">
            <span class="text-underline-dashed" matTooltip="Missense Z">
              Mis Z
            </span>
          </mat-header-cell>
          <mat-cell *matCellDef="let element" class="number">
            <span *ngIf="element.gnomad && element.gnomad.mis">
              {{ element.gnomad.mis.z.toFixed(2) }}
            </span>
            <span *ngIf="!element.gnomad || !element.gnomad.mis" class="na">NA</span>
          </mat-cell>
        </ng-container>

        <ng-container matColumnDef="gnomadLofZ">
          <mat-header-cell *matHeaderCellDef class="number">
            <span class="text-underline-dashed" matTooltip="LoF pLI">
              LoF pLI
            </span>
          </mat-header-cell>
          <mat-cell *matCellDef="let element" class="number">
            <span *ngIf="element.gnomad && element.gnomad.lof">
              {{ element.gnomad.lof.pLI.toFixed(2) }}
            </span>
            <span *ngIf="!element.gnomad || !element.gnomad.lof" class="na">NA</span>
          </mat-cell>
        </ng-container>

        <ng-container matColumnDef="dgvGain">
          <mat-header-cell *matHeaderCellDef class="number border-left">
            Total # Gain
          </mat-header-cell>
          <mat-cell *matCellDef="let element" class="number border-left">
            <span *ngIf="element.dgv">
              {{ element.dgv.gains }}
            </span>
            <span *ngIf="!element.dgv" class="na">NA</span>
          </mat-cell>
        </ng-container>

        <ng-container matColumnDef="dgvLoss">
          <mat-header-cell *matHeaderCellDef class="number">
            Total # Loss
          </mat-header-cell>
          <mat-cell *matCellDef="let element" class="number">
            <span *ngIf="element.dgv">
              {{ element.dgv.losses }}
            </span>
            <span *ngIf="!element.dgv" class="na">NA</span>
          </mat-cell>
        </ng-container>

        <ng-container *matHeaderRowDef="displayedColumns">
          <div class="mat-header-row header-group-row border-top">
            <div class="mat-header-cell" role="columnheader"></div>
            <div class="mat-header-cell header-group border-left text-center justify-content-center" role="columnheader" style="flex: 2">OMIM</div>
            <div class="mat-header-cell header-group border-left text-center justify-content-center" role="columnheader" style="flex: 4">ClinVar</div>
            <div class="mat-header-cell border-left text-center justify-content-center" role="columnheader" style="flex: 3">Geno2MP</div>
            <div class="mat-header-cell border-left text-center justify-content-center" role="columnheader" style="flex: 3">gnomAD</div>
            <div class="mat-header-cell header-group border-left text-center justify-content-center" role="columnheader" style="flex: 2">DGV</div>
          </div>
          <mat-header-row class="align-items-stretch"></mat-header-row>
        </ng-container>
        <mat-row *matRowDef="let row; columns: displayedColumns" class="align-items-stretch"></mat-row>
      </mat-table>

      <div *ngIf="loading">
        <div class="table-row-loading-wrapper w-100" *ngFor="let gene of genes.slice(vFrom, vTo)">
          <div class="table-row-loading w-100">
            <div class="table-cell-loading">
              {{ gene }}
            </div>
          </div>
        </div>
      </div>

      <mat-paginator
        [length]="genes.length"
        [pageSize]="gPerPage"
        [pageSizeOptions]="[10, 30, 50]"
        [pageIndex]="curPage"
        (page)="onPageChange($event)"
      ></mat-paginator>
    </div>
  </div>
</div>