<div class="loading-fullscreen" *ngIf="geneLoading" [@fadeInOut]>
  Loading ...
</div>

<mat-sidenav-container>
  <mat-sidenav #sidenav class="sidenav" mode="side" position="end" opened disableClose
    [ngClass]="{ 'opened': sidenavOpened, 'closed': !sidenavOpened }"
  >
    <app-sidenav
      [variant]="variantString"
      [gene]="gene"
      (change)="toggleSidenav($event)"
    ></app-sidenav>
  </mat-sidenav>

  <mat-sidenav-content id="human-result-container"
    [ngClass]="{ 'sidenav-opened': sidenavOpened, 'sidenav-closed': !sidenavOpened }"
  >
    <div id="TOP"></div>
    <app-navbar showSearch="true" showMenuButton="true"></app-navbar>

    <div *ngIf="gene !== null || variant" class="container-fluid py-5 px-md-5">

      <!-- Gene, Input information starts -->
      <div class="row">
        <div class="col-12">
          <div class="title" *ngIf="gene || variant">
            <h1 *ngIf="geneEntrezId && gene">{{ gene.symbol }}</h1>
            <h1 *ngIf="geneEntrezId && variant">&nbsp;/&nbsp;</h1>
            <h1 *ngIf="variant">{{ variantString }}</h1>
          </div>
          <div class="title" *ngIf="proteinInput">
            <h1>{{ proteinInput }}</h1>
          </div>
        </div>

        <div class="col-12 mt-3" *ngIf="geneCandidates">
          <mat-form-field style="margin-bottom: -1.25em">
            <mat-label>Genes contain {{ variantString }}</mat-label>
            <mat-select [value]="gene" (selectionChange)="onGeneSelectionChange($event)">
              <mat-option *ngFor="let aGene of geneCandidates" [value]="aGene">
                {{ aGene.symbol }}
              </mat-option>
            </mat-select>
          </mat-form-field>
        </div>

        <div class="col-12 subtitle mt-4">
          <h4 *ngIf="gene" [@fadeInOut]>
            <span class="label">Name</span>
            {{ gene.name }}
          </h4>
          <h4 *ngIf="gene && gene['alias']" [@fadeInOut]>
            <span class="label">Aliases</span>
            {{ gene.alias.join(', ') }}
          </h4>
        </div>

        <!-- OMIM text description start -->
        <div class="col-12 mt-3" id="OMIMText"
          [ngClass]="{ 'skeleton': omimLoading }"
          *ngIf="geneLoading || omimLoading || omimData" [@toggle]
        >
          <h4 class="m-0 title" *ngIf="omimData">OMIM Description of {{ gene.symbol }}</h4>
          <p class="mt-3 mb-0" *ngIf="omimData">{{ omimData['title'] || '' }}</p>

          <p class="e2e-intter-html-bound mt-3 mb-0"
            [innerHTML]="omimData['description'] || ''" *ngIf="omimData"
          ></p>
        </div>
        <!-- OMIM text description ends -->

      </div>
      <!-- Gene, Input information ends -->

      <h2 class="mt-5 text-uppercase">Disease Data</h2>

      <!-- OMIM starts -->
      <div class="row section mt-3" id="OMIM">
        <div class="col-12" [ngClass]="{ 'skeleton': omimLoading }">
          <app-omim
            [gene]="gene"
            [loading]="omimLoading"
            [data]="omimData"
          ></app-omim>
        </div>
      </div>
      <!-- OMIM ennds -->

      <!-- dbNSFP starts -->
      <div class="row section mt-5" id="DbNSFP" *ngIf="variant">
        <div class="col-12">
          <app-dbnsfp [variant]="variant"></app-dbnsfp>
        </div>
      </div>
      <!-- dbNSFP ends -->

      <!-- ClinVar starts -->
      <div class="row section mt-5" id="ClinVar">
        <div class="col-12">
          <app-clinvar [gene]="gene"></app-clinvar>
        </div>
      </div>
      <!-- ClinVar ends -->

      <!-- Geno2MP starts -->
      <div class="row section mt-5" id="Geno2MP">
        <div class="col-12">
          <app-geno2mp
            [gene]="gene"
            [variant]="variant"
          ></app-geno2mp>
        </div>
      </div>
      <!-- Geno2MP ends -->

      <!-- DECIPHER starts -->
      <div class="row section mt-5" id="DECIPHER">
        <div class="col-12">
          <app-decipher-disease
            [gene]="gene"
            [variant]="variant"
          ></app-decipher-disease>
        </div>
      </div>
      <!-- DECIPHER ends -->

      <mat-divider class="mt-5"></mat-divider>

      <h2 class="mt-5 text-uppercase">Control Data</h2>

      <!-- gnomAD starts -->
      <div class="row section mt-3" id="GnomAD">
        <div class="col-12">
          <h3 class="mb-0 d-inline-block align-middle">gnomAD</h3>
          <a class="mb-0 d-inline-block ml-3 align-middle"
            *ngIf="gene && gene.xref && gene.xref.ensemblId"
            href="https://gnomad.broadinstitute.org/gene/{{ gene.xref.ensemblId }}"
            target="_blank"
          >
            <button mat-button color="primary">
              <mat-icon class="mr-1 icon-18px">open_in_new</mat-icon>
              More on gnomAD
            </button>
          </a>
        </div>

        <!-- gnomAD variant -->
        <div class="col-12 mt-3" *ngIf="variant">
          <h4 class="mt-0 mb-3">
            Population Allele Frequencies of {{ variantString }} from gnomAD
          </h4>
          <app-gnom-ad
            [variant]="variant"
          ></app-gnom-ad>
        </div>

        <!-- gnomAD gene -->
        <div class="col-12 mt-3" *ngIf="gene">
          <h4 class="mt-0 mb-3">Control Population Summary of {{ gene.symbol }} from gnomAD</h4>
          <app-gnom-ad-gene
            [gene]="gene"
          ></app-gnom-ad-gene>
        </div>
      </div>
      <!-- gnomAD ends -->

      <!-- DGV starts -->
      <div class="row section mt-5" id="DGV">
        <div class="col-12">
          <app-dgv
            [variant]="variant"
            [gene]="gene"
          ></app-dgv>
        </div>
      </div>
      <!-- DGV ends -->

      <!-- DECIPHER Control starts -->
      <div class="row section mt-5" id="DECIPHER-Control" *ngIf="variant">
        <div class="col-12">
          <app-decipher
            [gene]="gene"
            [variant]="variant"
          ></app-decipher>
        </div>
      </div>
      <!-- DECIPHER Control ends -->

      <div class="row section mt-5" id="phenotypes">
        <div class="col-12">
          <app-phenotypes
            [gene]="gene"
            [orthologs]="orthologs"
          ></app-phenotypes>
        </div>
      </div>

      <div class="row section mt-5" id="gene-ontology">
        <div class="col-12">
          <app-gene-ontology
            [gene]="gene"
            [orthologs]="orthologs"
          ></app-gene-ontology>
        </div>
      </div>

      <!-- Alignment starts -->
      <div class="row section mt-5" id="alignment">
        <div class="col-12">
          <app-diopt-alignment
            [gene]="gene"
          ></app-diopt-alignment>
        </div>
      </div>
      <!-- Alignment ends -->

    </div>
  </mat-sidenav-content>
</mat-sidenav-container>

<app-scroll-top-button
  [right]="sidenavOpened ? '262px' : '84px'"
></app-scroll-top-button>